domain/application/state/application.go

```diff
@@ -99,17 +100,18 @@ func (st *State) CreateIAASApplication(
 	name string,
 	args application.AddIAASApplicationArg,
 	units []application.AddUnitArg,
-) (coreapplication.ID, error) {
+) (coreapplication.ID, []coremachine.Name, error) {
```

Comment:
Wouldn't it be better to return the machine's ID here instead of the name? it's consistent to the application, and we usually return UUIDs.